crates/eframe/src/epi.rs

```diff
@@ -472,6 +472,10 @@ pub struct WebOptions {
     /// Configures wgpu instance/device/adapter/surface creation and renderloop.
     #[cfg(feature = "wgpu")]
     pub wgpu_options: egui_wgpu::WgpuConfiguration,
+
+    /// Configures if the render pass should use a depth buffer.
+    #[cfg(feature = "wgpu")]
+    pub depth_texture: bool,
```

Comment:
maybe this should be in `WgpuConfiguration` since it only affects wgpu?

Reply:
it would actually make much more sense to have a `Option<wgpu::TextureFormat>` passed in which then internally is constrained to be a depth format. That's a short comin of the `winit.rs` frontend of egui-wgpu right now.
That said, there is the slightly bigger task here to move the multisampling and depth_buffer options from native options over to something that is used both for web and native.

Reply:
Done in the latest commit. A minor papercut I ran into that's worth considering is `WgpuConfiguration::default()`. With the current setup, the following code will result in a mostly opaque error (due to using the wrong backend):
```rust
let options = WebOptions {
  wgpu_options: {
    depth_format: Some(wgpu::TextureFormat::Depth32Float),
    ..Default::default()
  },
  ..Default::default()
};
```

It might be worth having a separate `#[cfg(target = "wasm32")] impl Default for WgpuConfig`.